common/src/main/java/whocraft/tardis_refined/registry/DeferredRegistry.java

```diff
@@ -15,12 +17,7 @@ public abstract class DeferredRegistry<T> {
     /** Register using a Supplier */
     public abstract <R extends T> RegistrySupplier<R> register(String id, Supplier<R> supplier);
 
-    /** Create a RegistrySupplierHolder. This has similar behaviour to a Supplier but has vanilla Holder attributes such as use in tags*/
-    public abstract <I extends T> RegistrySupplierHolder<T, I> registerHolder(final String name, final Supplier<I> sup);
-
-    /** Get the underlying registry, which includes all entries added by any mod that has a DeferredRegistry with the same ResourceKey.
-     * All lookup methods should be called from here.*/
-    public abstract Registry<T> getRegistry();
```

Comment:
Forge's registry system in 1.20.1 still retains an older design which was completely isolated from vanilla. The inability to access the underlying vanilla registry make this method unfeasible unfortunately. Instead, we need to expose lookup methods such as keySet and entrySet manually, so that a common interface is present for both Fabric and Forge experiences.